ons-spark/spark-analysis/big-data-workflow.md

```diff
@@ -0,0 +1,118 @@
+
+- It is a good idea to agree on a coding plan including code style for the project and Git branch strategies in advance.
+
+- Ensure code is readable, well documented, and follows a logical structure.
+
+### 3. Use [code reviews](https://best-practice-and-impact.github.io/qa-of-code-guidance/peer_review.html) 
+- Peer review of code should be used to quality assure and improve your code. A fresh pair of eyes may also be able to spot places where things could be done more efficiently to save computing resource.
+
+- This also helps other members of your team understand the code in detail and can help build capability in the team. It is important to avoid situations where a single developer is relied upon to understand, write and maintain all code as this can cause major problems if that person is away or leaves the team. 
+
+## **Data exploration and quality**
+
+### 4. Be aware of the size of your dataset
+- How many rows and columns does your dataset have?
+- If the dataset is relatively small (less than a few million rows) use Python or R for analysis instead of PySpark or SparklyR.
+- Consider also switching to using Python or R at points in your pipeline where data becomes small due to filtering or aggregation to save resource.
+- Is the size of the data likely to grow over time? If this is the case you should consider how you will minimise resource use as the amount of data to process increases.
+
+### 5. Reduce the size of your dataset to use fewer compute resources
+- Generally, it is unnecessary and not good practice to carry out the majority of the analysis stages outlined in figure 1 on a full dataset, especially if it is very large. 
+- Keep the size of your data in the EDA, detailed analysis and development stages as small as possible to conserve resource. 
+- Consider [taking samples of your data](../spark-functions/sampling), especially in the EDA phase.
+    - Use a simple [sample size calculator](https://www.calculator.net/sample-size-calculator.html) to estimate the size of the sample to take.
+    - Ideally, you should take a sample small enough that you can use Python or R for analysis instead of PySpark/SparklyR.
+    - You can validate your sample by comparing descriptive statistics to the full dataset.
+- Aggregate data where possible.
+    - This can be done in any stage of analysis and may even form part of your regular production pipeline.
+    - You can also use aggregation to help find edge cases that might not be captured by sampling your data. This can be particularly useful in the EDA phase to ensure that these extreme cases are not missed in the development of your pipeline.
+- Consider what data you really need and filter this out as early as possible to reduce the size of your data. 
+
+### 6. Consider creating synthetic data to work on
+- Create minimal size dummy data containing edge cases to work with in the development phase.
```

Comment:
I know the Faker Python package is often recommended for creating synthetic data, and I think there's been one or two RDSA talks about it so it's definitely used in ONS. Might be a good idea to link to it here (I don't think you need to explain too much about it, just mention it might be a good package to help create synthetic data)

Reply:
Thanks for the suggestion! I've added a mention of Faker for Python and fakeR for R now with links to documentation. I'm not sure if you think it's worth linking directly to one of the RDSA talks instead though?